FontForge hanging when showing/hiding layer in high DPI environment (Linux)

### When reporting a bug/issue:
- [ ] Screenshot
- [x] The FontForge version and the operating system you're using
- [x] The behavior you expect to see, and the actual behavior
- [x] Steps to reproduce the behavior 
- [ ] \(optional) Possible solution/fix/workaround

**Version**
```
 Based on sources from 08:50 UTC 17-Feb-2019-ML-D.
 Based on source from git with hash: 
fontforge 08:50 UTC 17-Feb-2019
libfontforge 20190217
```
Running on Fedora/X11. I have a high-DPI screen; I set screen width to 26 inches, which makes the UI scale to work on the screen mostly fairly well. In the glyph viewer, however, trying to check or uncheck the visibility of a layer (which at this DPI is done by showing of an eye icon instead of a check box) causes FF to hang indefinitely.

To reproduce, run FF and set the screen width to 26 inches (resolution might need to be 3840x2160 as well). Then try to show/hide a layer.

@jtanx I believe so. 

And that must be it. I was effectively compiling in release mode before I figured out how to properly compile in debug mode. Easy enough to check so ... here is the message:
```
[194/312] Building C object gdraw/CMakeFiles/gdraw.dir/gradio.c.o
../gdraw/gradio.c: In function ‘gradio_expose’:
../gdraw/gradio.c:366:21: warning: iteration 6 invokes undefined behavior [-Waggressive-loop-optimizations]
  366 |             pts[i].x=.5*w*cos(angle)+x+w/2;
      |             ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
../gdraw/gradio.c:364:9: note: within this loop
  364 |         for (c=0, i=0; c<=6; c++) {
      |         ^~~
```
Given an invocation of "undefined behavior" on the part of the compiler, there's also the possibility that @ctrlcctrlv isn't seeing this because he's doing a debug compile and all this traces to aggressive optimization rather than the logic problem (which is not to excuse the latter).

--- a/gdraw/gradio.c
+++ b/gdraw/gradio.c
@@ -350,7 +350,7 @@ return( false );
 
     } else if ( (!gr->ison) && gr->onbox == &visibility_on_box ) {
          /* draw closed eye */
-        GPoint pts[6];
+        GPoint pts[7];
         int c,i;
         double angle;
 	int bp = gr->onbox->border_type==bt_none ? 0 : GDrawPointsToPixels(pixmap,gr->onbox->border_width);
@@ -367,7 +367,7 @@ return( false );
             pts[i].y=.5*h*sin(angle)+y+h/4;
 
              /* draw lashes */
-            if (i>0 && i<5) GDrawDrawLine(pixmap, pts[i].x,pts[i].y, .75*w*cos(angle)+x+w/2, .75*h*sin(angle)+y+h/4, fg);
+            if (i>0 && i<6) GDrawDrawLine(pixmap, pts[i].x,pts[i].y, .75*w*cos(angle)+x+w/2, .75*h*sin(angle)+y+h/4, fg);
             ++i;
         }
         GDrawDrawPoly(pixmap, pts, i, fg);

--- a/gdraw/gradio.c
+++ b/gdraw/gradio.c
@@ -351,7 +351,6 @@ return( false );
     } else if ( (!gr->ison) && gr->onbox == &visibility_on_box ) {
          /* draw closed eye */
         GPoint pts[7];
-        int c,i;
         double angle;
 	int bp = gr->onbox->border_type==bt_none ? 0 : GDrawPointsToPixels(pixmap,gr->onbox->border_width);
         int x=gr->onoffrect.x+bp;
@@ -361,15 +360,15 @@ return( false );
         Color fg = g->state==gs_disabled?g->box->disabled_foreground:
 			g->box->main_foreground==COLOR_DEFAULT?GDrawGetDefaultForeground(GDrawGetDisplayOfWindow(pixmap)):
 			g->box->main_foreground;
-        for (c=0, i=0; c<=6; c++, i++) {
-            angle=(30+c/6.*120)*M_PI/180;
+        for (int i = 0; i <= 6; i++) {
+            angle=(30+i/6.*120)*M_PI/180;
             pts[i].x=.5*w*cos(angle)+x+w/2;
             pts[i].y=.5*h*sin(angle)+y+h/4;
 
              /* draw lashes */
             if (i>0 && i<6) GDrawDrawLine(pixmap, pts[i].x,pts[i].y, .75*w*cos(angle)+x+w/2, .75*h*sin(angle)+y+h/4, fg);
         }
-        GDrawDrawPoly(pixmap, pts, i, fg);
+        GDrawDrawPoly(pixmap, pts, sizeof(pts)/sizeof(pts[0]), fg);
     }
 
     GDrawPopClip(pixmap,&old2);

--- a/gdraw/gradio.c
+++ b/gdraw/gradio.c
@@ -361,14 +361,13 @@ return( false );
         Color fg = g->state==gs_disabled?g->box->disabled_foreground:
 			g->box->main_foreground==COLOR_DEFAULT?GDrawGetDefaultForeground(GDrawGetDisplayOfWindow(pixmap)):
 			g->box->main_foreground;
-        for (c=0, i=0; c<=6; c++) {
+        for (c=0, i=0; c<=6; c++, i++) {
             angle=(30+c/6.*120)*M_PI/180;
             pts[i].x=.5*w*cos(angle)+x+w/2;
             pts[i].y=.5*h*sin(angle)+y+h/4;
 
              /* draw lashes */
             if (i>0 && i<6) GDrawDrawLine(pixmap, pts[i].x,pts[i].y, .75*w*cos(angle)+x+w/2, .75*h*sin(angle)+y+h/4, fg);
-            ++i;
         }
         GDrawDrawPoly(pixmap, pts, i, fg);
     }

